crates/rpc-types-eth/src/transaction/mod.rs

```diff
@@ -338,87 +338,89 @@ impl TryFrom<Transaction> for TxEnvelope {
     }
 }
 
-impl TransactionResponse for Transaction {
-    type Signature = Signature;
-
-    fn tx_hash(&self) -> B256 {
-        self.hash
+impl alloy_consensus::Transaction for Transaction {
+    fn chain_id(&self) -> Option<ChainId> {
+        self.chain_id
     }
 
     fn nonce(&self) -> u64 {
         self.nonce
     }
 
-    fn block_hash(&self) -> Option<BlockHash> {
-        self.block_hash
+    fn gas_limit(&self) -> u64 {
+        self.gas
     }
 
-    fn block_number(&self) -> Option<u64> {
-        self.block_number
+    fn max_fee_per_gas(&self) -> u128 {
+        self.max_fee_per_gas.unwrap_or(self.gas_price.unwrap_or_default())
     }
 
-    fn transaction_index(&self) -> Option<u64> {
-        self.transaction_index
+    fn max_priority_fee_per_gas(&self) -> u128 {
+        self.max_fee_per_gas.unwrap_or(self.gas_price.unwrap_or_default())
```

Comment:
will get cleaned up by manual impl of serialisation for the the `alloy_rpc_types_eth::Transaction` type. then we don't need to have one field each for the essentially mutually exclusive fields `max fee per gas` and `gas price`, we just need the field `max fee per gas`, and the manual serialisation impl will just check `if self.ty() < 2` then use `gasPrice` as label, otherwise use `maxBaseFee`